test/e2e/framework/job/wait.go

```diff
@@ -161,6 +191,16 @@ func isJobFailed(j *batchv1.Job) bool {
 	return false
 }
 
+func isJobCompleted(j *batchv1.Job) bool {
```

Comment:
I would suggest to commonize this function with `isJobFailed` as `hasConditionTrue` (or named alike). You could extract that change to separate commit in the PR.

Reply:
Done! I've created a `hasAnyConditionTrue` function which replaces all the `isJob...` functions since they were duplicating code. Let me know if this suits you 😊

Reply:
I liked the simplicity of invoking the checks before: `hasAnyConditionTrue(curr, []batchv1.JobConditionType{ batchv1.JobFailed})` vs `isJobFailed` :) 

How about keeping `isJobFailed`, `isJobComplete`, and `isJobFinished = isJobFailed || isJobComplete`? Then underneath have a helper function `isConditionTrue` which is used in `isJobFailed` and `isJobComplete`?